src/test/java/io/jenkins/plugins/analysis/warnings/recorder/FilesScannerITest.java

```diff
@@ -130,6 +138,92 @@ public void findIssuesWithMultipleFiles() {
         assertThat(result).hasErrorMessages("Skipping file 'zero_length_file.xml' because it's empty");
     }
 
+
+    /**
+     * Runs the {@link FilesScanner} on a workspace with multiple files where some do match the criteria.
+     *
+     * @see <a href="http://issues.jenkins-ci.org/browse/JENKINS-51588">Issue 51588</a>
+     */
+    @Test
+    public void findIssuesWithMultipleFilesReachableWithSymlinks() throws IOException {
+        FreeStyleProject project = createJobWithWorkspaceFile(SYMLINKS_WORKSPACE);
+
+        FilePath workspace = getWorkspace(project);
+        Path path = Paths.get(workspace.getRemote());
+        Path realPath = path.resolve("actual_files");
+
+        assertThat(realPath.toFile().exists()).isTrue();
+
+//        Path symlinkPath = path.resolve(Paths.get("subdir", "link_to_actual_files"));
+        Path subdirPath = path.resolve("subdir");
+        assertThat(subdirPath.toFile().mkdirs()).isTrue();
+
+        Path symlinkPath = subdirPath.resolve("link_to_actual_files");
+
+        try {
```

Comment:
extract these 5 lines into method and catch IO exception as well then test needs no `throws` anymore (reuse in next method)